src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java

```diff
@@ -130,11 +131,21 @@ public void run() {
                                     indexSettings.put("index.number_of_shards", 1);
                                     indexSettings.put("index.auto_expand_replicas", "0-all");
 
-                                    boolean ok = client.admin().indices().create(new CreateIndexRequest(opendistrosecurityIndex)
-                                            .settings(indexSettings))
-                                            .actionGet().isAcknowledged();
-                                    LOGGER.info("Index {} created?: {}", opendistrosecurityIndex, ok);
-                                    if(ok) {
+                                    boolean existsOrCreated = client.admin().indices()
```

Comment:
What is the benefit of checking if the index already exists? Now there will be possibly two requests instead of one when one seems to provide exactly the same functionality. The `create` request should not be much more heavyweight compared to the `does exist` request to try to avoid it.

Reply:
Do you mean to always try creating and catch the failure as index exists case? 
Only the exists condition is going to be executed after the index is created. 
IMO why do something that we know node will always cry about and will throw an error ?

Reply:
Is there already a check that the index does not exist on line 235 before `installDefaultConfig` is set to `true`? Is this additional check prevents race condition between nodes or it is a different check?

Reply:
ES resolves concurrency for document updates amongst the nodes via document versioning.
So that looks like some different check (maybe threads within the same node).

Reply:
Based on the comments the code on line 235 checks if the index exists (but does it using different API). What is the difference between using `clusterService.state().metaData().hasConcreteIndex(opendistrosecurityIndex)` and `client.admin().indices().exists(new IndicesExistsRequest(opendistrosecurityIndex))`? As an attempt to create the index would be done only if `installDefaultConfig` is `true`, so `clusterService.state().metaData().hasConcreteIndex(opendistrosecurityIndex)` must be `false`, what is the purpose of `client.admin().indices().exists(new IndicesExistsRequest(opendistrosecurityIndex))` check?

Reply:
IndicesExistsRequest uses admin client to check if index exists. Maybe at some point the index was not concrete index and that api was used or need admin client 🤔 I did try with `clusterService.state().metaData().hasConcreteIndex(opendistrosecurityIndex)` and it returns false

Reply:
I still don't understand the difference between using admin client and cluster service state metadata (Yes, I understand that the former one is a network call while the last one uses in-memory state. But should both be consistent and if not, why one will be different than the other?)

In addition, `exists()` and `IndicesExistsRequest` are removed from the latest ES. IMO, we should try to avoid introducing a dependency on it now.

Please check other ES plugins to see how they check whether or not an index exists and create it.